server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -106,13 +113,19 @@ public void afterRefresh(boolean didRefresh) {
                                 .max(Comparator.comparingLong(IndexFileNames::parseGeneration));
 
                             if (latestSegmentInfos.isPresent()) {
+                                Set<String> segmentFilesFromSnapshot = new HashSet<>(refreshedLocalFiles);
                                 refreshedLocalFiles.addAll(SegmentInfos.readCommit(storeDirectory, latestSegmentInfos.get()).files(true));
                                 segmentInfosFiles.stream()
                                     .filter(file -> !file.equals(latestSegmentInfos.get()))
                                     .forEach(refreshedLocalFiles::remove);
 
                                 boolean uploadStatus = uploadNewSegments(refreshedLocalFiles);
                                 if (uploadStatus) {
+                                    if (segmentFilesFromSnapshot.equals(new HashSet<>(refreshedLocalFiles))) {
+                                        segment_info_snapshot_filename = uploadSegmentInfosSnapshot(latestSegmentInfos.get(), segmentInfos);
+                                        refreshedLocalFiles.add(segment_info_snapshot_filename);
```

Comment:
The `uploadSegmentInfosSnapshot` seems to be uploading the file to `remoteDirectory`, why do we need to add it to  `refreshedLocalFiles` (and apparently re-upload again with `remoteDirectory.uploadMetadata`)?

Reply:
`remoteDirectory.uploadMetadata()` uploads only the metadata file which keeps the mapping of files uploaded to remote store.

Filename is added to `refreshedLocalFiles` so that metadata file knows about snapshot file uploaded to remote store.